addons/mail/static/src/models/message/message.js

```diff
@@ -442,6 +442,30 @@ function factory(dependencies) {
                 this.originThread.isModeratedByCurrentPartner
             );
         }
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsSubjectEqualThreadName() {
+            if (!this.subject || !this.originThreadName) {
+                return false;
+            }
+            const prefix = ['re:', 'fw:', 'fwd:'];
+            let cleanedSubject = this.subject.toLowerCase();
+            let threadName = this.originThreadName.toLowerCase();
+            for (const pref of prefix) {
+                if (threadName === cleanedSubject) {
+                    return true;
+                }
+                if (cleanedSubject.startsWith(pref)) {
+                    cleanedSubject = cleanedSubject.replace(pref, '').trim();
+                }
+                if (threadName.startsWith(pref)) {
+                    threadName = threadName.replace(pref, '').trim();
+                }
```

Comment:
With the `return true` approach, I don't think we need to clean the thread name anymore. Maybe just adding the trim after the lowercase at the start.

Feel free to add more tests for covering specific cases with thread being named "re: test" and subject being "test" (should be different), "re: test" (should be equal), "fw: re: test" (should be equal) for example.